### PR TITLE
Success flow + modal changes

### PR DESCRIPTION
**Pull request by tgfassina**
- hid a bunch of elements from UI and left only those that have an implemented behavior
- replace sliding up panel with RN modal
- ability to get to the end of the flow and execute onSuccess
- modal closes at the end

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: src/ConnectWithVault.tsx [hunk 1]
```diff
@@ -46,30 +53,23 @@ const ConnectWithVault: React.FC<Props> = (props: Props) => {
         lightTheme={true}
       />
 
-      {props.displayTooltip && (
```
Comment (hvhallmann) [inline src/ConnectWithVault.tsx:49]: not sure why to remove this part of the screen.
It is not possible the user is going to reach this state if something fails?

**Review by hvhallmann — COMMENTED** [review]
Looks good overall,
could remove some comments,
And it is not clear how we can connect this with the Beam sign in flow.

File: src/components/TextButton.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
```
[the remark this answers is not in the document]
Comment (tgfassina) [reply src/components/TextButton.tsx:1]: It's a component for this button here:
image

File: src/ConnectWithVault.tsx (hunk 1 above)
Comment (tgfassina) [reply src/ConnectWithVault.tsx:49]: it's because we're not checking password yet. that's probably coming back soon

**Pull request merged**